### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by celery6**
minor adjustments to hopefully increase certainty that logger won't overflow any buffers or get stuck.
- give back logwritemutex before exiting logGeneric (I thought josh fixed this already but I couldn't find that change?)
- get rid of counting the log header string separately from the log msg. It's all under MAX_MSG_LENGTH now for simplicity and less confusing/potentially incorrect logic.
- change fullBuffersQueue to length NUM_LOG_BUFFERS instead of `NUM_LOG_BUFFERS - 1`. reasoning described in comment
- replace stupid source enums with source strings

### CODE REVIEW

**Comment by celery6** [comment]
testing code:
- enabled controller, log, flightphase tasks.
- Added log calls in controller and flightphase. 
- Added uart print in log task. 

Verify that the logs are in fact printing

image

**Comment by Joe-Joe-Joe-Joe** [comment]
Nice
I can probably find time to test this in person tmr

**Review by Lucifersan — APPROVED** [review]
works great other than the bits of code i had to remove
idk if its great practice to like approve after i just committed smth but it does seem to be working!
it still does the thing that it did in the last line of christine's ss where it gets halfway through a message and then it just cuts off with "[other message]", I'm not sure if thats what this is trying to resolve? Given how christine didn't really flag that I'm assuming not but I will not merge this for now for that reason
it is logging tho and otits runs and passes and controler absolutely spams messages and also its mispelled

**Review by Lucifersan — CHANGES_REQUESTED** [review]
nvm lol the formatting is actually maybe an issue do not merge till that is fixed

**Comment by celery6** [comment]
@Lucifersan just realized the test print uses uart_transmit not printf_ so this does actually print the entire buffer together.

This issue is actually just the hal_uart_transmit timeout of 500 ticks being too low since unsurprisingly it takes longer than 0.5s to transmit a book-length novel of logs :skull: 

increased timeout to 3000 and it works as expected. (log buffer itself is actually fine.) we can ignore this issue

**Comment by celery6** [comment]
ill clean up the testing code then this should be good to merge!

**Comment by celery6** [comment]
removed testing code, also cleaned up commits. lgtm?

**Review by Lucifersan — APPROVED** [review]
does not lgtmwtf i didnt see that

**Comment by Lucifersan** [comment]
mommy what.
im gonna change that and test it lmao
Screenshot 2024-06-28 at 3 28 17 PM

**Comment by celery6** [comment]
wtf I literally built and it built how tf 😭😭

**Pull request merged**
